cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -275,118 +249,136 @@ endforeach()
 if(DEFINED ENV{ARROW_AWSSDK_URL})
   set(AWSSDK_SOURCE_URL "$ENV{ARROW_AWSSDK_URL}")
 else()
-  set(AWSSDK_SOURCE_URL
-      "https://github.com/aws/aws-sdk-cpp/archive/${ARROW_AWSSDK_BUILD_VERSION}.tar.gz")
+  set_urls(
+    AWSSDK_SOURCE_URL
+    "https://github.com/ursa-labs/thirdparty/releases/download/latest/aws-sdk-cpp-${ARROW_AWSSDK_BUILD_VERSION}.tar.gz"
+    "https://github.com/aws/aws-sdk-cpp/archive/${ARROW_AWSSDK_BUILD_VERSION}.tar.gz"
+    "https://dl.bintray.com/ursalabs/arrow-awssdk/aws-sdk-cpp-${ARROW_AWSSDK_BUILD_VERSION}.tar.gz/aws-sdk-cpp-${ARROW_AWSSDK_BUILD_VERSION}.tar.gz"
```

Comment:
Can we remove duplicated `aws-sdk-cpp-${ARROW_AWSSDK_BUILD_VERSION}.tar.gz` path?

Reply:
I'll look into simplifying this path in https://issues.apache.org/jira/browse/ARROW-6407